File: src/main/java/com/vizientinc/dungeonbase/models/Item.java
package com.vizientinc.dungeonbase.models;

import com.vizientinc.dungeonbase.requests.ItemRequest;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Item {
    @Id
    String id;
    String name;
    String description;
    String location;

    public Item(ItemRequest itemRequest){
        this.name = itemRequest.getName();
        this.description = itemRequest.getDescription();
        this.location = itemRequest.getLocation();
    }
}

File: dotnet/Models/Item.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace dungeonbase.Models
{
    public class Item
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [BsonElement("id")]
        public string Id { get; set; }

        [BsonElement("name")] 
        public string Name { get; set; }

        [BsonElement("description")] 
        public string Description { get; set; }

        [BsonElement("location")] 
        public string Location { get; set; }
    }
}